client/apis/volumegroupsnapshot/v1beta1/types.go

```diff
@@ -354,16 +345,6 @@ type VolumeGroupSnapshotContentStatus struct {
 	VolumeSnapshotHandlePairList []VolumeSnapshotHandlePair `json:"volumeSnapshotHandlePairList,omitempty" protobuf:"bytes,6,opt,name=volumeSnapshotHandlePairList"`
 }
 
-// PVVolumeSnapshotContentPair represent a pair of PV names and
-// VolumeSnapshotContent names
-type PVVolumeSnapshotContentPair struct {
-	// PersistentVolumeRef is a reference to the persistent volume resource
-	PersistentVolumeRef core_v1.LocalObjectReference `json:"persistentVolumeRef,omitempty" protobuf:"bytes,1,opt,name=persistentVolumeRef"`
-
-	// VolumeSnapshotContentRef is a reference to the volume snapshot content resource
-	VolumeSnapshotContentRef core_v1.LocalObjectReference `json:"volumeSnapshotContentRef,omitempty" protobuf:"bytes,2,opt,name=volumeSnapshotContentRef"`
-}
-
 // VolumeGroupSnapshotContentSource represents the CSI source of a group snapshot.
```

Comment:
Not visible in PR - Can `VolumeHandle` and `SnapshotHandle` inside `VolumeSnapshotHandlePair` be empty? Or they both have to be specified. I didn't see these fields marked as `required` and neither see any validation kinda logic preventing saving of these fields without any values. Correct me if I missed something.

Reply:
VolumeSnapshotHandlePair itself is not required, but if a VolumeSnapshotHandlePair is specified, then both VolumeHandle and SnapshotHandle should be required. Otherwise, it is not very useful.
We can add CEL for them.

Reply:
> We can add CEL for them.
Yeah we should IMO.

Reply:
@leonardoce Can you make changes to make VolumeHandle and SnapshotHandle required in VolumeSnapshotHandlePair?

Reply:
Sure! I think it's better to file a different PR for that and keep this as a giant substitution. If we use this PR to review the API, it will become really difficult to rebase

Reply:
Discussed with @gnufied.  He prefers this change to be made as part of this PR.

Reply:
The main reason I wanted all API changes in same PR is, so as we don't have to do follow up API fixes.

Reply:
@gnufied I see both fields are already marked as required (meaning they are not optional) in the CRD.
See:

https://github.com/kubernetes-csi/external-snapshotter/blob/b79aef97261427cf2765410dc7f9d6647b7b3928/client/config/crd/groupsnapshot.storage.k8s.io_volumegroupsnapshotcontents.yaml#L290-L312

Do you still need me to add some CEL rules for them? Thanks!

Reply:
That should be sufficient.  The API server will reject it if VolumeHandle or SnapshotHandle is not set in VolumeSnapshotHandlePair.

Reply:
Can you add "// Required." as comments for both VolumeHandle and SnapshotHandle in types.go?

Reply:
I did it and rebased the PR too.